fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -338,11 +338,11 @@ private FHIRRestOperationResponse doPatchOrUpdate(String type, String id, FHIRPa
 
                     // If the id of the input resource is different from the id of the search result,
                     // then throw exception.
-                    if (newResource.getId() != null && newResource.getId() != null
-                            && !newResource.getId().equalsIgnoreCase(id)) {
+                    if (newResource.getId() != null && id != null && !newResource.getId().equalsIgnoreCase(id)) {
```

Comment:
do we consistently ignore case when working with id values?

Reply:
In this particular case, I left it the way it was because I assume it was put there for good reason. We can certainly verify that equalsIgnoreCase is the right behavior there.

Reply:
From here:
http://hl7.org/fhir/http.html#general
```
Note: All URLs (and ids that form part of the URL) defined by this specification are case sensitive.
```

Reply:
I will update the PR.